Source/Dafny/Compilers/Compiler-cpp.cs

```diff
@@ -2424,15 +2424,22 @@ protected class ClassWriter : IClassWriter {
       Contract.Assert(assemblyLocation != null);
       var codebase = System.IO.Path.GetDirectoryName(assemblyLocation);
       Contract.Assert(codebase != null);
-      var warnings = "-Wall -Wextra -Wpedantic -Wno-unused-variable -Wno-deprecated-copy -Wno-unused-label";
-      if (RuntimeInformation.IsOSPlatform(OSPlatform.Linux) || RuntimeInformation.IsOSPlatform(OSPlatform.Windows)) {
-        warnings += " -Wno-unused-but-set-variable";
-      } else {
-        warnings += " -Wno-unknown-warning-option";
-      }
-      var args = warnings + $" -g -std=c++17 -I {codebase} -o {ComputeExeName(targetFilename)} {targetFilename}";
+      var args = new List<string> {
+                       "-Wall",
+                       "-Wextra",
+                       "-Wpedantic",
+                       "-Wno-unused-variable",
+                       "-Wno-deprecated-copy",
+                       "-Wno-unused-label",
+                       "-Wno-unused-but-set-variable",
+                       "-Wno-unknown-warning-option",
+                        "-g",
+                        "-std=c++17",
+                        "-I", codebase,
+                        "-o", ComputeExeName(targetFilename),
+                        targetFilename };
       compilationResult = null;
-      var psi = new ProcessStartInfo("g++", args) {
+      var psi = new ProcessStartInfo("g++", string.Join(" ", args)) {
```

Comment:
You need to use [ArgumentList](https://docs.microsoft.com/en-us/dotnet/api/system.diagnostics.processstartinfo.argumentlist?view=net-6.0#system-diagnostics-processstartinfo-argumentlist), otherwise you have the same problem as before with strings (broken escaping / injection)

Reply:
Ah! That's much nicer.